pkg/gather/interface.go

```diff
@@ -1,11 +1,16 @@
 package gather
 
 import (
-	"context"
-
-	"github.com/openshift/insights-operator/pkg/recorder"
+	"github.com/openshift/insights-operator/pkg/gather/common"
 )
 
+// Interface is an interface for gathering types
 type Interface interface {
-	Gather(context.Context, []string, recorder.Interface) error
+	// GetName returns the name of the gatherer
+	GetName() string
+	// ListGatheringFunctions lists names of the functions implemented by current gatherer
+	ListGatheringFunctions() []string
```

Comment:
Wouldn't a slice of `common.GatheringFunction` make more sense as the return value?

Reply:
Hm returning the list of `common.GatheringClosure` should work